src/coreclr/vm/method.inl

```diff
@@ -6,6 +6,30 @@
 #ifndef _METHOD_INL_
 #define _METHOD_INL_
 
+inline bool MethodDesc::IsEligibleForTieredCompilation()
+{
+    LIMITED_METHOD_DAC_CONTRACT;
+
+#ifdef FEATURE_TIERED_COMPILATION
+    _ASSERTE(GetMethodDescChunk()->DeterminedIfMethodsAreEligibleForTieredCompilation());
+    return (VolatileLoadWithoutBarrier(&m_wFlags3AndTokenRemainder) & enum_flag3_IsEligibleForTieredCompilation) != 0;
+#else
+    return false;
+#endif
+}
+
+inline bool MethodDesc::IsEligibleForTieredCompilation_NoCheckMethodDescChunk()
+{
+    LIMITED_METHOD_CONTRACT;
+
+    // Just like above, but without the assert. This is used in the path which initializes the flag.
```

Comment:
Why does this matter? It seems very odd to have two different functions that do the same thing except for an debug only assert.

Reply:
At a minimum the assert one should call the one without the assert.